src/System.Text.Json/src/System/Text/Json/Serialization/ReferenceHandlingOnDeserialize.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Text.Json
+{
+    /// <summary>
+    /// This enum defines the various ways the <see cref="Utf8JsonReader"/> can deal with comments.
+    /// </summary>
+    public enum ReferenceHandlingOnDeserialize
+    {
+        /// <summary>
+        /// Indicates to the deserialier to ignore metadata properties in the payload.
+        /// </summary>
+        IgnoreMetadata,
```

Comment:
I believe earlier discussions is that we would have a default value of "Error" which would throw on deserialize and of course not emit metadata on serialize.

Reply:
I don't recall said discussion about Error for Deserialize, I do recall that we would throw by default when serializing a refence loop.

Reply:
If IgnoreMetadata is the default, does it throw when there are cycles (like today?)

Reply:
I think you are confussing Serialize with Deserialize, infinite loops or cycles are not a problem while **Deserializeing/Reading** because you are reading a string which is finite. And as of today, there is no way to have a cycle in a json payload.

The cycles are just a problem when you are **Serializing/Writing** because without a handling you would never finish writing your object graph.

Reply:
Sorry I had a typo -- I meant serialize. Yes the infinite loop only occurs on serialize today. 

I assume this enum also applies to serialize (whether or not metadata is included on serialize) but has "Deserialize" in its name (unless I'm missing some serializer-only setting).

Since this new feature enables a graph, which can have cycles, we need to ensure that valid or malformed\crafted JSON never results in an infinite loop on deserialize.

Reply:
> I assume this enum also applies to serialize

It does not, there will be a serialize-specific enum and a property on JsonSerializerOptions (probably called `WriteReferenceHandling`) meant for Serialization only. I will update the gist that contains the spec in a moment, sorry for any confussion.